src/core/thread/mle_router.cpp

```diff
@@ -735,51 +735,63 @@ void MleRouter::HandleLinkRequest(RxInfo &aRxInfo)
     aRxInfo.mClass = RxInfo::kPeerMessage;
     ProcessKeySequence(aRxInfo);
 
-    SuccessOrExit(error = SendLinkAccept(aRxInfo, neighbor, requestedTlvList, challenge));
+    if (aRxInfo.mMessageInfo.GetSockAddr().IsMulticast())
+    {
+        mDelayedSender.ScheduleLinkAccept(info, Random::NonCrypto::GetUint16InRange(0, kMaxLinkAcceptDelay));
+    }
+    else
+    {
+        error = SendLinkAccept(info);
+    }
 
 exit:
     LogProcessError(kTypeLinkRequest, error);
+    OT_UNUSED_VARIABLE(neighbor);
```

Comment:
This seems unnecessary? Because besides the `OPENTHREAD_CONFIG_MULTI_RADIO` and `OPENTHREAD_CONFIG_TIME_SYNC_ENABLE` blocks, `neighbor` is also used. For example, `neighbor->SetState(Neighbor::kStateLinkRequest);` which is not guarded by any build options.

Reply:
We need it. The `scan-build` check (pat of the `build` workflow on Github actions) did complain about "set but unused" `neighbor` variable. This build check does not enable either `MULTI_RADIO` or `TIME_SYNC_ENABLE`.